### ISSUE TITLE
[9.0.0] Data population don't work properly while HyperFormula is enabled 

### ISSUE THREAD

**Issue by aninde** [handsontable/handsontable#8050]
### Description
Autofill and validation sometimes don't work while HyperFormula is enabled with numeric cell type.
[beforeAutofill](https://handsontable.com/docs/8.4.0/Hooks.html#event:beforeAutofill) returns different value then in the same version without HF.

#### Reproduction:
1. Open demo.
2. Select cell `Mercedes A 160` (A1).
3. Autofill to the last most right cell.

#### Actual result

`beforeAutofill` had empty data, and makes nulls
<img width="414" alt="Screenshot 2021-05-17 at 21 15 41" src="https://user-images.githubusercontent.com/30186818/118544508-c7601c80-b755-11eb-9a92-b178180afbd1.png">

```
beforeAutofill:  
{from: {…}, to: {…}}
from:
col: 1
row: 0
__proto__: Object
to:
col: 3
row: 0
__proto__: Object
__proto__: Object

```
#### Expected result
`beforeAutofill` return correct data.
Proper values are rendering. Validation works and makes numeric cells red after pasting strings into them.
Working with older version of formulas https://jsfiddle.net/aninde/1gp2baw9/ 
<img width="499" alt="Screenshot 2021-05-17 at 21 12 29" src="https://user-images.githubusercontent.com/30186818/118543581-b95dcc00-b754-11eb-99d6-fdacf1144398.png">

```
"beforeAutofill: "
{
  col: 1,
  row: 0
}, {
  col: 3,
  row: 0
}, [["Mercedes A 160"]]
```

### Demo

https://jsfiddle.net/aninde/1gp2baw9/ (works) v8.4.0 + old enginge
https://codesandbox.io/s/validators-forked-j47sn v

### Your environment
* Handsontable version: v9.0.0 from [PR ](https://github.com/handsontable/handsontable/pull/7648)+ HF v.0.6.0
* Browser Name and version: Chrome 91
* Operating System: macOS

**Comment by AMBudnik** [comment]
### Case 2 for autofill
Demo: https://jsfiddle.net/handsoncode/3hd8tnkp/
1. Select header A:B
2. Autofill to right

![error](https://user-images.githubusercontent.com/10757813/118959206-65283700-b962-11eb-94c3-cc4bf933271b.gif)

**Result**
![image](https://user-images.githubusercontent.com/10757813/118958924-1f6b6e80-b962-11eb-9b5b-03faa772a821.png)

When we try to do anything else like selection or edit table UI breaks, no data is saved.

![broken](https://user-images.githubusercontent.com/10757813/118959270-74a78000-b962-11eb-8c54-0efb1eb758a7.gif)

The issue is gone when we comment out

```js
formulas: {
    engine: HyperFormula
}
```

**Comment by wszymanski** [comment]
It seems that there is one more regression. Width of headers aren't counted properly when `type` option isn't set for columns.

![autofill](https://user-images.githubusercontent.com/141330/119133986-dd632b00-ba3c-11eb-9d4c-79fb054c4312.gif)

For me, it's outside the scope of the task. It may be related to `AutoColumnSize` plugin. @aninde coul you confirm it and create an issue when necessary?

**Comment by budnix** [comment]
> It seems that there is one more regression. Width of headers aren't counted properly when `type` option isn't set for columns.
> 
> ![autofill](https://user-images.githubusercontent.com/141330/119133986-dd632b00-ba3c-11eb-9d4c-79fb054c4312.gif)
> 
> For me, it's outside the scope of the task. It may be related to `AutoColumnSize` plugin. @aninde could you confirm it and create an issue when necessary?

I'm already working on it https://github.com/handsontable/handsontable/issues/8063 :)

**Comment by aninde** [comment]
Both cases are fixed by todays build of v9.0.0 https://jsfiddle.net/aninde/4sn2obqf/

![Screenshot 2021-05-26 at 13 40 22](https://user-images.githubusercontent.com/30186818/119654620-f3039680-be28-11eb-994e-9bc192ae3b7d.png)

![Kapture 2021-05-26 at 13 46 48](https://user-images.githubusercontent.com/30186818/119654591-eb43f200-be28-11eb-931e-3a70cdebb764.gif)
